src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -108,6 +133,23 @@ public static bool ReportWithoutCrashUnlessCanceled(Exception exception)
             return ReportWithoutCrash(exception);
         }
 
+        /// <summary>
+        /// Use in an exception filter to report a non-fatal error. 
+        /// Calls <see cref="NonFatalHandler"/> unless the operation has been cancelled. 
+        /// The exception isn't passed through (the method returns true).
+        /// </summary>
+        /// <returns>True to catch the exception.</returns>
+        [DebuggerHidden]
+        public static bool ReportWithoutCrashUnlessCanceled(Exception exception, CancellationToken cancellationToken)
```

Comment:
nit: Should this be `ReportUnlessCancelledWithoutCrash`? Current name makes it seem like canceled state may impact if crash happens or not.

Reply:
We have been using `ReportUnlessCancelledWithoutCrash` for a while - it's not a new method. I'm only adding an overload. Should we decide to rename it it'd be a in different PR. I'm not sure the other order would help much - `ReportUnlessCancelledWithoutCrash` can also be interpreted as report unless "cancelled without crash".